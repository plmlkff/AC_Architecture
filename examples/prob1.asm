MAX: word 999
SUM: word 0
VAL: word 0
S_VAL: word 0
START: ld VAL
loop_three:
    ld VAL
    add #3
    cmp MAX
    jg #end
    wr VAL
    cmp S_VAL
    je #loop_five
    add SUM
    wr SUM
    ld VAL
    cmp S_VAL
    jg #loop_five
    jmp #loop_three
loop_five:
    ld S_VAL
    add #5
    cmp MAX
    jg #loop_three ; Переход на цикл тройки, так как она будет дорастать медленнее
    wr S_VAL
    cmp VAL
    je #loop_three
    add SUM
    wr SUM
    ld S_VAL
    cmp VAL
    jg #loop_three
    jmp #loop_five
end:
    ld SUM
    hlt