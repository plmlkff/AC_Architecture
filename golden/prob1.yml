in_source: |
  NUMBER: word 999
  SUM: word 0
  LOOP: ld #3
      push
      call #check_div
      cmp #1
      je #make_sum
      ld #5
      push
      call #check_div
      cmp #1
      je #make_sum
      ld NUMBER
      sub #1
      je #end
      wr NUMBER
      jmp #LOOP
  make_sum: ld SUM
      add NUMBER
      wr SUM
      ld NUMBER
      sub #1
      je #end
      wr NUMBER
      jmp #LOOP
  end:
      ld SUM
      hlt
  DIVIDER: word 0
  check_div: pop
      swap
      wr DIVIDER
      cmp #0
      je #bad_end
      ld NUMBER
  loop: sub DIVIDER
      jl #bad_end
      je #good_end
      jmp #loop
  good_end: ld #1
      ret
  bad_end: ld #0
      ret

in_stdin: |-
  5Trash
out_code: |-
  [999, 0, {"name": "ld", "op_code": "LD", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 3}, {"name": "push", "op_code": "PUSH", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, {"name": "call", "op_code": "CALL", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 28}, {"name": "cmp", "op_code": "CMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 1}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 17}, {"name": "ld", "op_code": "LD", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 5}, {"name": "push", "op_code": "PUSH", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, {"name": "call", "op_code": "CALL", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 28}, {"name": "cmp", "op_code": "CMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 1}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 17}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "sub", "op_code": "SUB", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 1}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 25}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "jmp", "op_code": "JMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 2}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 1}, {"name": "add", "op_code": "ADD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 1}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "sub", "op_code": "SUB", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 1}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 25}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "jmp", "op_code": "JMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 2}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 1}, {"name": "hlt", "op_code": "HLT", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, 0, {"name": "pop", "op_code": "POP", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, {"name": "swap", "op_code": "SWAP", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 27}, {"name": "cmp", "op_code": "CMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 0}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 40}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 0}, {"name": "sub", "op_code": "SUB", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 27}, {"name": "jl", "op_code": "JL", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 40}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 38}, {"name": "jmp", "op_code": "JMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 34}, {"name": "ld", "op_code": "LD", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 1}, {"name": "ret", "op_code": "RET", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, {"name": "ld", "op_code": "LD", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 0}, {"name": "ret", "op_code": "RET", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
out_stdout: |
  code.o
  ============================================================
  
  Ticks count: 21283307
out_log: |
  DEBUG   Machine:tick_log      Tick #21283269: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 5; DR: 25; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=25); IP: 25; AR: 25; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283270: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 5; DR: 25; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=25); IP: 25; AR: 25; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283271: MEMORY ACCESS TICK mIP: 1; AC: 0; BR: 5; DR: 25; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=25); IP: 25; AR: 25; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283272:  mIP: 2; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=25); IP: 25; AR: 25; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283273:  mIP: 3; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 25; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283274:  mIP: 4; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 25; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283275:  mIP: 5; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283276:  mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283277: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283278: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283279: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283280: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283281: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283282: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283283: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283284: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283285: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283286: MEMORY ACCESS TICK mIP: 18; AC: 0; BR: 5; DR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283287:  mIP: 19; AC: 0; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283288:  mIP: 24; AC: 0; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283289:  mIP: 25; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283290:  mIP: 69; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 25; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283291:  mIP: 70; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283292:  mIP: 0; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 1; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283293:  mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283294: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283295: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283296: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283297: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283298: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283299: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283300: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283301: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283302: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283303: MEMORY ACCESS TICK mIP: 1; AC: 233168; BR: 5; DR: 233168; SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283304:  mIP: 2; AC: 233168; BR: 5; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='ld', op_code=<OpCode.LD: 24>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=1); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283305:  mIP: 3; AC: 233168; BR: 5; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283306:  mIP: 19; AC: 233168; BR: 5; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 26; AR: 26; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #21283307:  mIP: 68; AC: 233168; BR: 5; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 26; AR: 26; N: False; Z: True
